src/workingtitle-vcockpits-instruments-cj4/html_ui/Pages/VCockpit/Instruments/Airliners/CJ4/FMC/CJ4_FMC_InitRefIndexPage.js

```diff
@@ -1102,51 +1102,80 @@ class CJ4_FMC_InitRefIndexPage {
+        let freqText = "---.--";
+        let gsAngleText = "-.--°";
+        let locTrueBrgText = "---T";
+        
+        let rightSelectionKeyProc = undefined;
 
         let destination = fmc.flightPlanManager.getDestination();
-        if (destination) {
-            destinationString = destination.ident;
-            if (destination.infos) {
-                destinationString = destinationString + "/" + destination.infos.name;
-            }
-        }
-        
-        if (fmc.flightPlanManager.getApproach()) {
+        if (!destination) {
+            CJ4_FMC_InitRefIndexPage.ShowPage25ErrorMsg(fmc, "NO DEST SELECTED", "FPLN");
+            rightSelectionKeyProc = CJ4_FMC_RoutePage.ShowPage1;
+        } else {
             let approach = fmc.flightPlanManager.getApproach();
-            approachName = fmc.flightPlanManager.getApproach().name;
-            approachFrequency = fmc.flightPlanManager.getApproachNavFrequency();
-        }
-        if (fmc.flightPlanManager.getApproachRunway()) {
-            appRunway = fmc.flightPlanManager.getApproachRunway();
-        }
+            if (!approach.name && !fmc.vfrLandingRunway) {
+                CJ4_FMC_InitRefIndexPage.ShowPage25ErrorMsg(fmc, "NO APPROACH SELECTED", "ARRIVAL");
+                rightSelectionKeyProc = CJ4_FMC_DepArrPage.ShowArrivalPage;
+            } else {
+                arrAirportText = destination.ident;
+                if (destination.infos) {
+                    arrAirportText = arrAirportText + "/" + destination.infos.name;
+                }
 
-        let appRunwayDirection = new Number(appRunway.direction);
-        let appRunwayElevation = new Number(appRunway.elevation * 3.28);
+                approachText = approach.name || "RW" + Avionics.Utils.formatRunway(fmc.vfrLandingRunway.designation);
+                let approachRunway = fmc.flightPlanManager.getApproachRunway() || fmc.vfrLandingRunway;
+                rwyThresholdAltText = Math.trunc(approachRunway.elevation * 3.28) + " FT";
+                if (approach.name.startsWith("ILS")) {
+                    freqText = fmc.flightPlanManager.getApproachNavFrequency().toFixed(2);
+                    locTrueBrgText = Math.trunc(approachRunway.direction).toString().padStart(3, "0") + "T";
+                    gsAngleText = "3.00°";
+                }
+                
+                fmc._templateRenderer.setTemplateRaw([
+                    [" ACT[blue]", "ARRIVAL DATA  [blue]"],
+                    [" ARR AIRPORT[blue]"],
+                    [arrAirportText],
+                    [" APPR[blue]", "FREQ [blue]"],
+                    [approachText, freqText],
+                    [" GS ANGLE[blue]"],
+                    [gsAngleText],
+                    [" LOC TRUE BRG[blue]"],
+                    [locTrueBrgText],
+                    [" RWY THRESHOLD ALT[blue]"],
+                    [rwyThresholdAltText],
+                    ["-----------------------[blue]"],
```

Comment:
I don't work much on FMS, but there is an FMCsperator constant I think for these dashes which could be nice to add instead of manual -'s. Just a thought.

Reply:
There is but we don't really use it as my FMC renderer doesn't take it